RIOT-OS/RIOT
# sys: preprocessor successor module

### Contribution description



Generate a header file, containing successor definitions up to a fixed maximum.
This allows to use the preprocessor to compute the successor of a number and is useful to construct priority values for `auto_init`.
This is similar to [boost preprocessor arithmetik](https://www.boost.org/doc/libs/1_43_0/boost/preprocessor/arithmetic/inc.hpp). 

### Testing procedure



See usage in `tests/external_module_dirs/external_modules/external_module/external_module.c`
```
CFLAGS+=-DCONFIG_AUTO_INIT_ENABLE_DEBUG=1 BOARD=nucleo-f767zi make flash term

2022-07-06 09:39:31,837 # auto_init: ztimer_init (1010)
2022-07-06 09:39:31,840 # auto_init: auto_init_random (1040)
2022-07-06 09:39:31,844 # auto_init: auto_init_external_module (1041)
2022-07-06 09:39:31,847 # auto_init: auto_init_miot_mtd (1249)
2022-07-06 09:39:31,851 # auto_init: auto_init_miot_node (1359)
2022-07-06 09:39:31,860 # main(): This is RIOT! (Version: 2022.07-devel-788-g87bda-preprocessor_successor)
2022-07-06 09:39:31,862 # Message: Linking worked
2022-07-06 09:39:31,863 # Initialization worked!
```

### Issues/PRs references



#18003
